controllers/kubeadmconfig_controller.go

```diff
@@ -76,29 +76,9 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, nil
 	}
 
-	// Find the owner reference
-	// The cluster-api machine controller set this value.
-	var machineRef *v1.OwnerReference
-	for _, ref := range config.OwnerReferences {
-		if ref.Kind == machineKind.Kind && ref.APIVersion == machineKind.GroupVersion().String() {
-			machineRef = &ref
-			break
-		}
-	}
-	if machineRef == nil {
-		log.Info("did not find matching machine reference")
-		return ctrl.Result{}, nil
-	}
-
-	// Get the machine
-	machine := &capiv1alpha2.Machine{}
-	machineKey := client.ObjectKey{
-		Namespace: req.Namespace,
-		Name:      machineRef.Name,
-	}
-
-	if err := r.Get(ctx, machineKey, machine); err != nil {
-		log.Error(err, "failed to get machine")
+	machine, err := util.GetOwnerMachine(ctx, r.Client, config.ObjectMeta)
+	if err != nil {
+		log.Error(err, "could not get owner machine")
```

Comment:
Should there be a check here for a `nil` machine? similar to the check included here: https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/173/files

Reply:
yes! I missed that one `return nil, nil`, good catch